cmd/publishing-bot/github.go

```diff
@@ -65,28 +66,32 @@ func ReportOnIssue(e error, logs, token, org, repo string, issue int) error {
 		return fmt.Errorf("failed to comment on issue #%d: HTTP code %d", issue, resp.StatusCode)
 	}
 
-	// re-open issue in case it was closed
-	_, resp, err = client.Issues.Edit(ctx, org, repo, issue, &github.IssueRequest{
-		State: github.String("open"),
-	})
-	if err != nil {
-		return fmt.Errorf("failed to re-open issue #%d: %v", issue, err)
-	}
-	if resp.StatusCode >= 300 {
-		return fmt.Errorf("failed to re-open issue #%d: HTTP code %d", issue, resp.StatusCode)
-	}
-
 	// delete all other comments from this user
-	comments, resp, err := client.Issues.ListComments(ctx, org, repo, issue, &github.IssueListCommentsOptions{})
+	comments, resp, err := client.Issues.ListComments(ctx, org, repo, issue, &github.IssueListCommentsOptions{
+		ListOptions: github.ListOptions{PerPage: 100},
+	})
 	if err != nil {
 		return fmt.Errorf("failed to get github comments of issue #%d: %v", issue, err)
 	}
 	if resp.StatusCode != http.StatusOK {
 		return fmt.Errorf("failed to get github comments of issue #%d: HTTP code %d", issue, resp.StatusCode)
 	}
 	for _, c := range comments {
-		if *c.User.ID == *myself.ID && *c.ID != *newComment.ID {
-			resp, err = client.Issues.DeleteComment(ctx, org, repo, *c.ID)
+		if *c.User.ID != *myself.ID {
+			glog.Infof("Skipping comment %d not by me, but %v", *c.ID, c.User.Name)
+			continue
+		}
+		if *c.ID == *newComment.ID {
+			continue
+		}
+
+		glog.Infof("Deleting comment %d", *c.ID)
+		resp, err = client.Issues.DeleteComment(ctx, org, repo, *c.ID)
+		if err != nil {
+			return fmt.Errorf("failed to delete github comment %d of issue #%d: %v", *c.ID, issue, err)
+		}
+		if resp.StatusCode >= 300 {
```

Comment:
shouldn't we follow the redirection on 3xx? iirc, it's the default behavior of golang http client.

Reply:
yes, in a follow-up this would be fine.